### PR TITLE
POST: User Game Score

### LINKED ISSUE #31: POST to Back End Creates `user_game` column in db, along with associated columns in `answered_questions`
```json
{ 
  "data": { 
    "type": "game",
    "id": 12,
    "date": "09/12/23", 
    "user_id": 12
    "catgories": [{
      "category": "before & after",
      "category_emoji": :),
      "questions": [{
          "answer": "a putting Green Eggs and Ham",
          "question": "Golf short game practice area served up in the title of a Dr. Seuss book",
          "points": 1
          "user_response": "green eggs"
          "result": self_corrected
        },
        {
          "answer": "the Hebrew National League",
          "question": "If they used a well-known brand of kosher hot dogs instead of bats, the Dodgers & Phillies would be in this baseball group",
          "points": 2,          
          "user_response": "green eggs"
          "result": self_corrected
        },
        {
          "answer": "the Cheshire Cat scan",
          "question": "Grinning enigmatic acquaintance of Alice in Wonderland seen in cross section by doctors",
          "points": 3
          "user_response": "green eggs"
          "result": self_corrected
      }]
    },{
      "category": "before & after",
      "category_emoji": :),
      "questions": [{
          "answer": "a putting Green Eggs and Ham",
          "question": "Golf short game practice area served up in the title of a Dr. Seuss book",
          "points": 1
          "user_response": "green eggs"
          "result": self_corrected
        },
        {
          "answer": "the Hebrew National League",
          "question": "If they used a well-known brand of kosher hot dogs instead of bats, the Dodgers & Phillies would be in this baseball group",
          "points": 2
          "user_response": "green eggs"
          "result": self_corrected
        },
        {
          "answer": "the Cheshire Cat scan",
          "question": "Grinning enigmatic acquaintance of Alice in Wonderland seen in cross section by doctors",
          "points": 3
          "user_response": "green eggs"
          "result": self_corrected
      }]
    },{
      "category": "before & after",
      "category_emoji": :),
      "questions": [{
          "answer": "a putting Green Eggs and Ham",
          "question": "Golf short game practice area served up in the title of a Dr. Seuss book",
          "points": 1
          "user_response": "green eggs"
          "result": self_corrected
        },
        {
          "answer": "the Hebrew National League",
          "question": "If they used a well-known brand of kosher hot dogs instead of bats, the Dodgers & Phillies would be in this baseball group",
          "points": 2
          "user_response": "green eggs"
          "result": "incorrect"
        },
        {
          "answer": "the Cheshire Cat scan",
          "question": "Grinning enigmatic acquaintance of Alice in Wonderland seen in cross section by doctors",
          "points": 3
          "user_response": "green eggs"
          "result": "correct"
      }]
    }] 
  }
}
```

### PR DESCRIPTION
**Pull request by ethanrossblack**
## Description
This PR does A LOT. The basic and primary part is that it implements the `POST` endpoint to save a user's game and answers to the BE database.

## Post request

To post the result of a user's game, the endpoint is now `POST`, `/api/user_games`

The `POST` request should have the following structure: 

```json
{ 
  "data": { 
    "type": "user_games",
    "user_id": 12,
    "game_id": 1,
    "score": 6,
    "user_answers": [
      {
        "game_question_id": 1,
        "user_answer": "putting green eggs and ham",
        "result": "self_corrected"
      },
      {
        "game_question_id": 2,
        "user_answer": "the Hebrew National league",
        "result": "correct"
      },
      {
        "game_question_id": 3,
        "user_answer": "cheshire cat scan",
        "result": "self_corrected"
      },
      {
        "game_question_id": 4,
        "user_answer": "coors",
        "result": "correct"
      },
      {
        "game_question_id": 5,
        "user_answer": "",
        "result": "skipped"
      },
      {
        "game_question_id": 6,
        "user_answer": "jack daniels",
        "result": "incorrect"
      },
      {
        "game_question_id": 7,
        "user_answer": "a down town",
        "result": "self_corrected"
      },
      {
        "game_question_id": 8,
        "user_answer": "happy pappy",
        "result": "incorrect"
      },
      {
        "game_question_id": 9,
        "user_answer": "a low blow",
        "result": "self_corrected"
      }
    ]
  }
}
```

A successful `POST` returns the following `201` response body:

```json
{
  "data"=>{
    "id"=>"8", 
    "type"=>"user_game", 
    "attributes"=>{
      "user_id"=>12, 
      "score"=>6, "created_at"=>"2023-09-20T20:27:59.247Z"
      }, 
    "relationships"=>{
      "game"=>{
        "data"=>{
          "id"=>"16", 
          "type"=>"game"}
      }
    }
  }
}
```

## Update Backend DB Schema

I had to update some of the table schema to make things work, so here is some detailed documentation for our updated schema. (ty ChatGPT for helping format these tables nicely)

* Renamed `answered_questions` to `user_questions`, which is more intuitive
* Added some connections between `user_answers` and `game_questions`

Untitled

### game_questions

| Column Name  | Data Type | Key       | Description |
|--------------|-----------|-----------|-------------|
| id           | varchar   | PK        |             |
| game_id      | bigint    | FK (games)|             |
| question_id  | bigint    | FK (questions)|    |
| created_at   | datetime  |           |             |
| updated_at   | datetime  |           |             |

### games

| Column Name  | Data Type | Key       | Description |
|--------------|-----------|-----------|-------------|
| id           | varchar   | PK        |             |
| orig_date    | string    |           |             |
| created_at   | datetime  |           |             |
| updated_at   | datetime  |           |             |

### questions

| Column Name  | Data Type | Key       | Description |
|--------------|-----------|-----------|-------------|
| id           | varchar   | PK        |             |
| clue         | string    |           |             |
| answer       | string    |           |             |
| category     | string    |           |             |
| clue_id      | integer   |           | references a clue_id on jservice.io            |
| value        | integer   |           |             |
| category_id  | integer   |           | refers to category id on jservice.io            |
| created_at   | datetime  |           |             |
| updated_at   | datetime  |           |             |

### user_answers

| Column Name      | Data Type | Key       | Description |
|------------------|-----------|-----------|-------------|
| id               | varchar   | PK        |             |
| user_game_id     | bigint    | FK (user_games)| |
| user_answer      | string    |           |             |
| result           | string    |           |             |
| created_at       | datetime  |           |             |
| updated_at       | datetime  |           |             |
| game_question_id | bigint    | FK (game_questions)| |

### user_games

| Column Name  | Data Type | Key       | Description |
|--------------|-----------|-----------|-------------|
| id           | varchar   | PK        |             |
| game_id      | bigint    | FK (games)|             |
| user_id      | integer   |           | references a user_id held on the FE            |
| created_at   | datetime  |           |             |
| updated_at   | datetime  |           |             |
| score        | integer   |           |             |

## Related Issue

closes #31 

## Checklist
- [x] I have reviewed the code changes.
- [x] I have tested this code.
- [x] I have updated the documentation.
- [x] I have checked for and resolved any merge conflicts.
- [x] I have assigned reviewers for this PR.
## Additional Notes
[Add any additional notes or context about this PR.]

I need some help with the enum when creating new `user_answers` in the `USerGamesController`

Could someone give that a look?

### CODE REVIEW

File: config/schedule.rb [hunk 1]
```diff
@@ -22,6 +22,6 @@
 # set :bundle_command, "/Users/ianlyell/.rbenv/shims/bundle exec"
 
 #Creates a new daily game in the DB at 1:00 AM everyday.
-every 1.minute do
+every 1.hour do
```
Comment (topher-nullset) [inline config/schedule.rb:25]: just making sure we are leaving this at every 1 hour.

File: db/migrate/20230920034829_change_answer_to_user_answer_in_user_answers.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class ChangeAnswerToUserAnswerInUserAnswers < ActiveRecord::Migration[7.0]
```
Comment (topher-nullset) [inline db/migrate/20230920034829_change_answer_to_user_answer_in_user_answers.rb:1]: good migrating, always forward!

File: db/seeds.rb [hunk 3]
```diff
@@ -8,6 +8,13 @@
 
 
 #Creates Questions in the database based on a category ID and a point value
+
+Game.delete_all
+Question.delete_all
+GameQuestion.delete_all
+UserGame.delete_all
```
Comment (topher-nullset) [inline db/seeds.rb:15]: just to be clear, we are deleting user game histories here, is this just until release?

File: app/services/j_api_service.rb [hunk 4]
```diff
@@ -1,7 +1,7 @@
 class JApiService
 
     def initialize
-        @min_date = "2000-01-01"
+        @min_date = "2001-11-26" # November 26th, 2001 is the date when the moetary values of first round Jeopardy questions switched from 100-500 to 200-1000.
```
Comment (topher-nullset) [inline app/services/j_api_service.rb:4]: This is the kind of secret knowledge that makes a hard problem easy!

File: db/seeds.rb (hunk 3 above)
Comment (ethanrossblack) [reply db/seeds.rb:15]: @ILyell  what was the thought process behind deleting all the data in the seed file?

**Review by topher-nullset — APPROVED** [review]
ok we wil push the questions forward

**Pull request merged**